### DISCUSSION TITLE
Renovate On-Premises: where to report a bug?

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#19679] [Request Help] [answered]
## Summary

- Where should users file a bug when they're using Renovate On-Premises?
- Explain in the bug report form for `renovatebot/renovate` if you're in the right place.

Maybe we should change things on the `whitesource/renovate` repository as well?

- Explain in the `whitesource/renovate` readme where to file bugs.
- Create bug report template for `whitesource/renovate` to point people to the right place.

### Bug report for On-Prem on `renovatebot/renovate`

A user filed a report using an old version of Renovate and I labeled the issue `retry latest version`, because we're on `34.x.x` and their version was `32.x.x`. The user replied: [^quote-user]

> We can't use the latest version. We are using Mend's Renovate on prem deployment helm chart which is using whitesource/renovate docker image and its latest version is 2.6.0 which is using Renovatebots 32.238.1 version.

So I removed the `retry latest version` again, after checking that `2.6.0` is the latest published version. 😄 

#### On-premises release cadence

Here's the Whitesource On-Premises docs for the release cadence: [^cadence]

> ## Releasing and Upgrading
> 
> The release cadence of WhiteSource Renovate is not fixed, as it will be determined largely by the importance and stability of new Renovate OSS features, which will typically be tested using the hosted Renovate GitHub App first.
> 
> When a new version of WhiteSource Renovate is pushed to Docker Hub, Release Notes will be added to this [github.com/whitesource/renovate](https://github.com/whitesource/renovate) repository.
> 
> Naturally, it is recommended that you use Renovate itself for detecting and updating WhiteSource Renovate versions if you are using a Docker Compose file internally for running WhiteSource Renovate.

### Edit bug report form?

The bug report form for the `renovatebot/renovate` repository [^bug-form] should explain if users are in the correct place for bugs with the Renovate On-premises version.

### Change files on the `whitesource/renovate` repository?

The `README.md` and the docs for `whitesource/renovate` are not clear to me about where I should file bugs with the On-Premises version. I'm willing to open a PR to change things, but I'll need the maintainers to say what to change. 😄 

[^quote-user]: https://github.com/renovatebot/renovate/issues/19677#issuecomment-1372109467
[^cadence]: https://github.com/mend/renovate-on-prem/blob/main/docs/overview.md#releasing-and-upgrading
[^bug-form]: https://github.com/renovatebot/renovate/blob/main/.github/ISSUE_TEMPLATE/bug_report.yml

**Comment by rarkins** [comment, marked as answer]
There is functionality which is exclusive to the On-premises container (e.g. webhook handling) while naturally most of the functionality is common with the Renovate OSS from this repo. If in doubt, users should post to the On-premises repo where they can be advised to raise an issue or discussion here instead. I also don't mind discussions in this repo, as we are pretty flexible on those.
